lib/options.go

```diff
@@ -187,6 +189,103 @@ func ParseCIDR(s string) (*IPNet, error) {
 	return &parsedIPNet, nil
 }
 
+// HostnameTrie is a tree-structured list of hostname matches with support
+// for wildcards exclusively at the start of the pattern. Items may only
+// be inserted and searched. Internationalized hostnames are valid.
+type HostnameTrie struct {
+	children map[rune]*HostnameTrie
+}
+
+// Regex description of hostname pattern to enforce blocks by. Global var
+// to avoid compilation penalty at runtime.
+// Matches against strings composed entirely of letters, numbers, or '.'s
+// with an optional wildcard at the start.
+//nolint:gochecknoglobals
+var legalHostnamePattern *regexp.Regexp = regexp.MustCompile(`^\*?(\pL|[0-9\.])*`)
+
+func legalHostname(s string) error {
+	if len(legalHostnamePattern.FindString(s)) != len(s) {
+		return errors.Errorf("invalid hostname pattern %s", s)
+	}
+	return nil
+}
+
+// UnmarshalJSON forms a HostnameTrie from the provided hostname pattern
+// list.
+func (t *HostnameTrie) UnmarshalJSON(data []byte) error {
```

Comment:
Unfortunately, you should also implement the `MarshalJSON()` method :disappointed: k6 needs to be able to marshal all of the options in a JSON format when it's making an [archive bundle](https://k6.io/docs/misc/archive-command). 

Try running `k6 archive --archive-out test.tar --block-hostname="*" github.com/loadimpact/k6/samples/http_get.js` and then `k6 run test.tar`. You'd get an error like this:
```
ERRO[0000] json: cannot unmarshal object into Go struct field Options.options.blockHostnames of type []string 
```

this is because the `metadata.json` in the archive will have `    "blockHostnames": {},` in it, which is not the array this code expects.

Reply:
Ah got it! Will go ahead and implement this. Is there anything else I've missed in my changes?

Reply:
I don't think so, having `MarshalJSON()` that produces an array should take care of the issue, since your code can already deal with arrays (after the envconfig issue is worked around). I think you don't need to use `k6 archive` every time though, `k6 inspect --block-hostname="*" github.com/loadimpact/k6/samples/http_get.js` should mostly do the same things but give you faster feedback.

Reply:
Or a unit test that makes an archive on the fly and runs it, for example take a look at https://github.com/loadimpact/k6/blob/4c39bcccf606aa3b4324ec5d4f3adfdfcf7d2fa5/js/runner_test.go#L179-L227

Or some of the other tests that make use of `Runner.MakeArchive()` and `Run()` the result: https://github.com/loadimpact/k6/blob/4c39bcccf606aa3b4324ec5d4f3adfdfcf7d2fa5/js/runner.go#L110-L112